src/ssmDocument/wizards/publishDocumentWizard.ts

```diff
@@ -3,400 +3,140 @@
+function createNamePrompter() {
+    return createInputBox({
+        title: localize('AWS.ssmDocument.publishWizard.ssmDocumentName.title', 'Name your document'),
+        buttons: createCommonButtons(),
+        validateInput: validateDocumentName,
+    })
 }
 
-export interface DocumentTypeQuickPickItem {
-    label: string
-    description?: string
-    detail?: string
-    documentType: SSM.DocumentType
+function createUpdateDocumentPrompter(region: string, documentType?: string) {
+    return createQuickPick(loadDocuments(region, documentType), {
+        title: localize(
+            'AWS.ssmDocument.publishWizard.ssmDocumentToUpdate.title',
+            'Select a document to update ({0})',
+            region
+        ),
+        noItemsFoundItem: {
+            label: localize(
+                'AWS.ssmDocument.publishWizard.ssmDocumentToUpdate.noDocument',
+                'No self-owned documents could be found. Please create and upload a Systems Manager Document before updating.'
+            ),
+            data: WIZARD_BACK,
+        },
+        buttons: createCommonButtons(),
+    })
 }
 
-export interface UpdateDocumentQuickPickItem {
-    label: string
-    description?: string
-    detail?: string
-    alwaysShow: boolean
-    name?: string
+function createPublishPrompter(region: string) {
+    const publishItems = [
+        {
+            label: localize('AWS.ssmDocument.publishWizard.publishAction.quickCreate.label', 'Quick Create'),
+            detail: localize(
+                'AWS.ssmDocument.publishWizard.publishAction.quickCreate.detail',
+                'Create a Systems Manager Document'
+            ),
+            data: PublishSSMDocumentAction.QuickCreate,
+        },
+        {
+            label: localize('AWS.ssmDocument.publishWizard.publishAction.quickUpdate.label', 'Quick Update'),
+            detail: localize(
+                'AWS.ssmDocument.publishWizard.publishAction.quickUpdate.detail',
+                'Update an existing Systems Manager Document'
+            ),
+            data: PublishSSMDocumentAction.QuickUpdate,
+        },
+    ]
+
+    return createQuickPick(publishItems, {
+        title: localize(
+            'AWS.ssmDocument.publishWizard.publishAction.title',
+            'Publish to AWS Systems Manager Document ({0})',
+            region
+        ),
+        buttons: createCommonButtons(),
+    })
 }
 
-export class DefaultPublishSSMDocumentWizardContext extends WizardContext implements PublishSSMDocumentWizardContext {
-    private documents: SSM.Types.DocumentIdentifierList | undefined
-
-    private readonly totalSteps: number = 3
+export class PublishSSMDocumentWizard extends Wizard<PublishSSMDocumentWizardResponse> {
+    public constructor(region?: string) {
+        super({ initState: { region } })
+        const form = this.form
 
-    public constructor(private readonly awsContext: AwsContext, private readonly regionProvider: RegionProvider) {
```

Comment:
There's a bunch of implementations passing around `awsContext` + `regionProvider` just to get regions. Not a good pattern to follow.